### PR TITLE
Fix for #21682 - Connection Pool bug occurs when Connection Validation is enabled and On Any Failure  Close All Connections option is used

### PR DESCRIPTION
**Pull request by srinivaskrishnan**
Fix for #21682 - Connection Pool bug occurs when Connection Validation is enabled and On Any Failure Close All Connections option is used

### CODE REVIEW

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

File: main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java [hunk 1]
```diff
@@ -879,10 +879,11 @@ protected ResourceHandle createSingleResourceAndAdjustPool(
         }
 
         ResourceHandle result = getNewResource(alloc);
-        if (result != null) {
-            alloc.fillInResourceObjects(result);
-            result.getResourceState().setBusy(true);
-        }
+        // The code below has been commented-out because it is effectively being run AGAIN by the caller
```
Comment (yaminikb) [inline main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java]: Please remove the commented code.

File: main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java [hunk 2]
```diff
@@ -1246,7 +1247,9 @@ private void incrementNumConnFailedValidation() {
     }
 
     private ResourceHandle getNewResource(ResourceAllocator alloc) throws PoolingException {
-        ds.addResource(alloc, 1);
+        // The wrapper method addResource() needs to be called instead of ds.addResource(), so that NumConnFree gets incremented after creating the new resource
```
Comment (yaminikb) [inline main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java]: Remove commented code

**Review by yaminikb — CHANGES_REQUESTED** [review]
Have you run the connectors dev tests?

File: main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java (hunk 1 above)
Comment (srinivaskrishnan) [inline main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java]: Done

File: main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java (hunk 2 above)
Comment (srinivaskrishnan) [inline main/appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java]: Done

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — CHANGES_REQUESTED** [review]
I had given this comment before - please remove commented code. It just clutters and impacts readability  of the code.

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Review by srinivaskrishnan — COMMENTED** [review]
Removed the commented code

**Comment by srinivaskrishnan** [comment]
@glassfishrobot Run CI tests please

**Pull request closed without merging**